File: C++/level2/숫자의_표현.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
    int answer = 0;
    int sum;
    int innerIndex;

    for(int i=1; i < n/2+1; i++){
        sum = 0;
        innerIndex = i;
        while(sum < n){
            sum += innerIndex;
            innerIndex++;
        }
        if(sum == n)
            answer++;
    }

    return answer+1;
}

File: JavaScript/level2/숫자의_표현.js
function solution(n) {
    var answer = 1;
    
    for(let i=1; i<n/2; i++){
        let sum = 0;
        let index = i;
        
        while(sum < n)
            sum += index++;
        
         if(sum === n)
            answer++;
    }
    
    return answer;
}